File: 0690-employee-importance/0690-employee-importance.cpp
/*
// Definition for Employee.
class Employee {
public:
    int id;
    int importance;
    vector<int> subordinates;
};
*/

class Solution {
public:
    void dfs(vector<Employee*> employees,int id ,int& sum,map<int,int> &IDs)
    {
        sum+=employees[IDs[id]]->importance;
        for(auto emp_id : employees[IDs[id]]->subordinates){
            dfs(employees,emp_id,sum,IDs);
        }
    }
    int getImportance(vector<Employee*> employees, int id) {

        // fill hashtable
        map<int,int> IDs;
        for(int i=0;i<(int)employees.size();++i){
            IDs[employees[i]->id]=i;
        }
        int sum=0;

        dfs(employees,id,sum,IDs);
        return sum;
    }
};

File: 0690-employee-importance/0690-employee-importance.cs
/*
// Definition for Employee.
class Employee {
    public int id;
    public int importance;
    public IList<int> subordinates;
}
*/

class Solution {
    Dictionary<int,Employee> id_to_emp=new Dictionary<int,Employee> ();
    
    private int DFS(int id){
        Employee emp = id_to_emp[id];
        int result=emp.importance;
        foreach(var sId in emp.subordinates){
            result+=DFS(sId);
        }
        return result;
    }
    public int GetImportance(IList<Employee> employees, int id) {
        for(int i=0;i<employees.Count;i++){
            id_to_emp.Add(employees[i].id,employees[i]);
        }
        
        return DFS(id);
    }
}